File: OpenCV/12_hough/python/main.py
import cv2, numpy as np
g=cv2.imread("input.jpg",cv2.IMREAD_GRAYSCALE); assert g is not None
e=cv2.Canny(g,80,160); lines=cv2.HoughLinesP(e,1,np.pi/180,80,minLineLength=50,maxLineGap=10); print("lineas",0 if lines is None else len(lines))

File: OpenCV/12_hough/cpp/main.cpp
#include <opencv2/opencv.hpp>
#include <iostream>
int main(){ auto g=cv::imread("input.jpg",cv::IMREAD_GRAYSCALE); if(g.empty()) return 1; cv::Mat e; cv::Canny(g,e,80,160); std::vector<cv::Vec4i> l; cv::HoughLinesP(e,l,1,CV_PI/180,80,50,10); std::cout<<"lineas "<<l.size()<<"\n"; }
